### PR TITLE
Add example for mapError

### PR DESCRIPTION
**Pull request by nitikagarw**
References #25468

### CODE REVIEW

**Comment by nitikagarw** [comment]
@johanandren @patriknw Please review.

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class MapError {
+
+  public static void main(String[] args) {
+
+    // #map-error
+
+    Source.from(Arrays.asList(-1, 0, 1))
+        .map(x -> 1 / x)
+        .mapError(
+            new PFBuilder<Throwable, Throwable>()
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java:21]: I think we should avoid the `PFBuilder` variant and use `.mapError(Class[E], Function[E, Throwable])` instead

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                .match(
+                    ArithmeticException.class,
+                    (ArithmeticException e) ->
+                        new UnsupportedOperationException(
+                            "Divide by Zero Operation is not supported"))
+                .build());
+    // #map-error
```
Comment (johanandren) [inline akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java:28]: Would be nice to "show" it actually having an effect, for example running with `Sink.seq` and then println in a `.whenComplete`

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java (hunk 1 above)
Comment (nitikagarw) [reply akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java]: Changed.

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java (hunk 2 above)
Comment (nitikagarw) [reply akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/MapError.java:34]: Done.

File: akka-docs/src/test/scala/docs/stream/operators/sourceorflow/MapError.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  implicit val system: ActorSystem = ActorSystem()
+  implicit val ec: ExecutionContext = system.dispatcher
+
+  //#map-error
+  Source(-1 to 1)
+    .map(1 / _)
+    .mapError {
+      case _: ArithmeticException =>
+        throw new UnsupportedOperationException("Divide by Zero Operation is not supported.") with NoStackTrace
```
Comment (johanandren) [inline akka-docs/src/test/scala/docs/stream/operators/sourceorflow/MapError.scala:24]: Not a big thing but the partial function is meant to return the `UnsupportedOperationException`, not throw it.

**Review by johanandren — COMMENTED** [review]
One small thing more, then it looks good to to me

File: akka-docs/src/test/scala/docs/stream/operators/sourceorflow/MapError.scala (hunk 3 above)
Comment (nitikagarw) [reply akka-docs/src/test/scala/docs/stream/operators/sourceorflow/MapError.scala]: Thanks for pointing out.

**Review by johanandren — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Thanks for contributing, @nitikagarw

**Pull request merged**